src/app/quick_views/media.cpp

```diff
@@ -0,0 +1,226 @@
+#include <QFrame>
+#include <QtCore/QVariant>
+#include <QtWidgets/QApplication>
+#include <QtWidgets/QComboBox>
+#include <QtWidgets/QDial>
+#include <QtWidgets/QHBoxLayout>
+#include <QtWidgets/QLabel>
+#include <QtWidgets/QPushButton>
+#include <QtWidgets/QSpacerItem>
+#include <QtWidgets/QVBoxLayout>
+#include <QtWidgets/QWidget>
+#include <QDebug>
+
+#include "app/arbiter.hpp"
+#include "app/widgets/dialog.hpp"
+#include "app/quick_views/media.hpp"
+#include "app/utilities/icon_engine.hpp"
+
+MediaQuickView::MediaQuickView(Arbiter &arbiter)
+    : QFrame()
+    , QuickView(arbiter, "Media", this)
+{
+
+}
+
+void MediaQuickView::init(){
+
+        AAHandler *aa_handler = arbiter.android_auto().handler;
+        mediaWidget = new MediaWidget(this->arbiter,this);
+
+        horizontalLayout = new QHBoxLayout(this);
+        horizontalLayout->setObjectName(QString::fromUtf8("horizontalLayout"));
+        horizontalLayout->setContentsMargins(0, 0, 0, 0);
+        mediaWidget = new MediaWidget(this->arbiter, this);
+        horizontalLayout->addWidget(mediaWidget);
+       
+        connect(aa_handler, &AAHandler::aa_media_playback_update, [this](const aasdk::proto::messages::MediaInfoChannelPlaybackData& playback){
+            mediaWidget->updatePlayback(playback);
+        });
+
+        connect(aa_handler, &AAHandler::aa_media_metadata_update, [this](const aasdk::proto::messages::MediaInfoChannelMetadataData& metadata){
+           mediaWidget->updateMetadata(metadata);
+        });
+
+}
+
+MediaWidget::MediaWidget(Arbiter &arbiter, QWidget *parent)
+    : QWidget(parent)
+    ,arbiter(arbiter)
+{
+
+    scroll_timer = new QTimer(this);
+    scroll_timer->start(33);//30fps
+    connect(scroll_timer, &QTimer::timeout, [this](){
+        
+        if(metadata_.has_track_name() && metadata_.has_artist_name()){
+            QString track_info(QString::fromStdString(metadata_.artist_name()) + 
+            " - " +
+            QString::fromStdString(metadata_.track_name()));
+            QFont myFont("arial", 10);
```

Comment:
id prefer if you used `Session::Forge::font` just to keep everything consistent and could have the size adjust based on a user's scale

Reply:
On the topic of scaling - is there going to be an issue with the hard coded constants I see?

Reply:
Will do! I'll also be sure to change it so the track information is centered according to the font metrics instead of the 5px offset like I have now. @icecube45 what do you mean by hard coded constants? Besides the font, I'm pretty sure everything is set in relation to the height and width of the widget, except the padding of course. If the user adjusts the scale, the height of the control bar changes because of the power buttons to the right, the MediaWidget draws the icons according to the height of the control bar so it adjusts to the scale automatically. I've attached some screenshots taken with a large and small scale applied. Let me know if I'm missing something else tho!
![large-scale](https://user-images.githubusercontent.com/63250719/166180821-1be04345-e2bb-4f7d-b0ae-d4f7897fd070.png)
![small-scale](https://user-images.githubusercontent.com/63250719/166180825-a6768973-ebcc-4dc7-ade1-21bb9ed0d21f.png)

Reply:
I was referring to things like the QSize(128,128) or drawArc(play_rect,1440,5760);, but im not too concerned

Reply:
i was a bit concerned when i saw that first but it seems like everything is done relative to the widgets size so things should size appropriately (the qsize doesnt matter much here since the painter determines the size and those constants there are just for the arc's angle so all good there)